src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,68 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final double MAX_SIZE = 100;
+    private static final int FIGURES_TYPES_COUNT = FigureType.values().length;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FIGURES_TYPES_COUNT);
+        FigureType figureType = FigureType.values()[index];
+
+        return switch (figureType) {
+            case CIRCLE -> getRandomCircle();
+            case SQUARE -> getRandomSquare();
+            case RECTANGLE -> getRandomRectangle();
+            case RIGHT_TRIANGLE -> getRandomRightTriangle();
+            case ISOSCELES_TRAPEZOID -> getRandomIsoscelesTrapezoid();
+            default -> getDefaultFigure();
+        };
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle("white", 10);
```

Comment:
The 'getDefaultFigure' method is returning a new Circle instance with hardcoded values. This can be improved for better code readability by assigning the values to variables before passing them to the constructor.